src/transformers/image_box_utils.py

```diff
@@ -0,0 +1,523 @@
+    if depth == 1 and len(boxes) < 4:
+        raise ValueError(f"Expected boxes to have at least 4 elements, got {len(boxes)} elements.")
+    if is_array_type(boxes) and _shape(boxes)[-1] < 4:
+        raise ValueError(f"Expected boxes to have at least 4 elements, got {_shape(boxes)[-1]} elements.")
+
+    # Base of recursion.
+    if depth == 2 and is_array_type(boxes):
+        return _convert_boxes_arrays_2d(boxes, input_format, output_format, image_size, check)
+
+    # Recursive approach.
+    elif depth == 1 and isinstance(boxes, (list, tuple) or is_array_type(boxes)):
+        boxes_2d = _expand_on_zero_dim(boxes)
+        return _convert_boxes_recursively(boxes_2d, input_format, output_format, image_size, check)[0]
+
+    elif depth == 2 and isinstance(boxes, (list, tuple)):
+        np_boxes = np.array(boxes)
+        np_converted_boxes = _convert_boxes_recursively(np_boxes, input_format, output_format, image_size, check)
+        converted_boxes = np_converted_boxes.tolist()
+        if isinstance(boxes, tuple):
+            converted_boxes = tuple([tuple(box) for box in converted_boxes])
+        return converted_boxes
+
+    elif depth == 3 and isinstance(boxes, (list, tuple)):
+        image_sizes = image_size if image_size is not None else [None] * len(boxes)
+        return type(boxes)(
+            [_convert_boxes_recursively(b, input_format, output_format, s, check) for b, s in zip(boxes, image_sizes)]
+        )
+
+    elif depth == 3 and is_array_type(boxes):
+        image_sizes = image_size if image_size is not None else [None] * len(boxes)
+        converted_boxes = [
+            _convert_boxes_recursively(b, input_format, output_format, s, check) for b, s in zip(boxes, image_sizes)
+        ]
+        return _stack(converted_boxes, axis=0)
+
+    else:
+        raise ValueError(f"Unsupported type of boxes: {type(boxes)}")
+
+
+def convert_boxes(
+    boxes: Union[ArrayType, List, Tuple],
+    input_format: str,
+    output_format: str,
+    image_size: Optional[Union[ArrayType, List, Tuple]] = None,
+    check: Optional[str] = "warn",
+):
+    """
+    Convert boxes from one format to another.
+
+    Bounding boxes can be provided as:
+        - A single box (torch.Tensor | np.ndarray | List[float, ...]): Box of shape (B,), where B is 4 or more elements,
+        and the first 4 elements are the coordinates of the box.
+        - Boxes from one image (torch.Tensor | np.ndarray | List[List[float, ...]]): A set of boxes (N, B) where N is
+        the number of boxes.
+        - Boxes from multiple images (torch.Tensor | np.ndarray | List[torch.Tensor | np.ndarray] | List[List[List[float, ...]]]):
+        A set of images with their boxes (I, N, B), where I is number of images. Can be represented as a single 3D array/tensor
+        or most likely as a list of 2D arrays/tensors.
+
+    Supported input/output bounding box formats:
+        - `absolute_xyxy` (aliases: `pascal_voc`, `xyxy`): [x_min, y_min, x_max, y_max]
```

Comment:
I'd rather we didn't try to handle different aliases and just keep it simple to e.g. `xyxy`